optimizely/event_builder.py

```diff
@@ -172,19 +182,39 @@ def _get_attributes(self, attributes):
     if not attributes:
       return []
 
-    for attribute_key in attributes.keys():
+    for attribute_key in sorted(attributes.keys()):
       attribute_value = attributes.get(attribute_key)
       # Omit falsy attribute values
       if attribute_value:
-        attribute = self.config.get_attribute(attribute_key)
-        if attribute:
+        # Check for reserved attributes
+        reserved_attrs = [ReservedAttributes.BUCKETING_ID, ReservedAttributes.USER_AGENT]
+        if attribute_key in reserved_attrs:
           params.append({
-            self.EventParams.EVENT_ID: attribute.id,
+            'entity_id': attribute_key,
             'key': attribute_key,
             'type': self.EventParams.CUSTOM,
-            'value': attribute_value,
+            'value': attribute_value
           })
 
+        else:
+          attribute = self.config.get_attribute(attribute_key)
+          if attribute:
```

Comment:
in case attribute is not found, pruning that attribute but where's the log? Add log or I would suggest revise it as per javascript code.